### ISSUE TITLE
onClickOutside ignored when a button is clicked. 

### ISSUE THREAD

**Issue by carpben** [Pomax/react-onclickoutside#286]
In [this](https://www.givingway.com/search.give) React-Redux search app the filter box is wrapped by OnClickOutside. 
![image](https://user-images.githubusercontent.com/13496178/43185707-5683afd4-8ff5-11e8-8a9d-873864d2eca8.png)
In general it works well. But when a different Filter Button is clicked it's filter box opens as well, and the onClickOutside is ignored. 
![image](https://user-images.githubusercontent.com/13496178/43185726-626dd194-8ff5-11e8-962a-b76ae293a602.png)
This is an undesired result. When there is a click outside, even on such a button, we'd like for the onClickOutside handler to be executed.  
There is no event.stopPropogation set on the filter button. We use onClickOutside with it's default settings. 
Any suggestions? Please let me know if you need more info. I can also add the map to the live site, so u can see the source code in the browser.

**Comment by carpben** [comment]
I think the best way is to add a dedicated URL for this purpose, which will include the search page plus the map file. This way you can view in Chrome dev tools both relevant components - the FilterBtn component, and the filter's DecisionBox component. Does this work for you?

**Comment by Pomax** [comment]
Honestly I'd rather have a real PoC, so something with maybe two components, of which the filter component is used several times, and shows off this interplay with exactly the minimum amount of code, stripped from any surrounding website-specific code like other components, libraries, CSS, etc. Trying to modify a running React application on an active site is not a good way to do debugging. 

Just copy the site to a new dir, remove anything that isn't the main container and the filters, remove all the CSS except CSS critical for the menu placement, replace any text with generic placeholders that are _just_ long enough to give visually understandable content, etc.

Basically if you're running into this, that minimizing exercise helps you figure out if this is you, or if this is the library. That should be sortable with a minimal example that I would be surprised surpasses 50 lines of code.

**Comment by carpben** [comment]
I made a siginficant effort to create a minimal version. I removed most libraries, all of Redux, and all other components. With deployment it took about 2 hours. 
https://bug-demo.firebaseapp.com/

I use a method called "atomic-design", which means the components structure is very modulare. And the CSS style is within the components, so removing the CSS from the components themselves will require lots of time. Hope you understand.

**Comment by carpben** [comment]
I used emotion-react to create styled components. The problem was that styled component of the filter btn, was created within the functional component of the FilterButton. Which basically means that everytime the FilterButton was rendered an new instance was created in the dom. After removing that 
StyledComponent out of the function, and just refering to it in the returned JSX the problem was solved.
